internal/scheduler/job.go

```diff
@@ -0,0 +1,63 @@
+package scheduler
+
+import (
+	"context"
+	"time"
+
+	"github.com/hashicorp/boundary/internal/errors"
+)
+
+// JobId is the unique id assigned to the job after registration, it is generated and
+// returned by RegisterJob.  The id is predictable and uses the job code and name as a seed.
+// This id is prefixed with "job_".
+type JobId string
+
+// Job defines an interface for jobs that can be invoked by the scheduler.
+type Job interface {
+	// Status reports the job’s current status.  The status is periodically persisted by
+	// the scheduler when a job is running, and will be used to verify a job is making progress.
+	Status() JobStatus
+
+	// Run performs the required work depending on the implementation.
+	// The context is used to notify the job that it should exit early.
+	Run(ctx context.Context) error
```

Comment:
Placeholder comment as I have a suggestion here but need to see implementation first.